source/edge.ts

```diff
@@ -1,2 +1,37 @@
+import { Node } from './node';
+
 export class Edge {
+  constructor(name: string, tail: Node, head: Node) {
+    this._name = name;
```

Comment:
This is a fine implementation for now, however at some point in the future edges won't belong to nodes. It will be possible, for example, to have an edge that starts or ends at a node but is detached. Think about a case where a user creates two nodes, S1 and S2 that are disconnected from each other... then they click to create an edge starting at S1, an edge will appear starting at S1 and following the mouse... the user can then move the mouse over S2 and click on S2 and only then will the edge has a tail of S2.